subsys/net/lib/download_client/src/http_download_client.c

```diff
@@ -482,6 +495,7 @@ int download_client_connect(struct download_client *const client, char *host,
 	}
 
 	client->host = host;
+	client->sec_tag = sec_tag;
```

Comment:
What happens if the first header received indicates disconnect? Will the `client->sec_tag` put into `download_client_connect(dl, dl->host, dl->sec_tag);`  be valid at that point?

I.E. should this be moved further up?

Reply:
Yes, the security tag is still valid. It can be considered constant for a given host, i.e. if you are downloading from the same host you will need the same security tag.